build_package.hs

```diff
@@ -0,0 +1,233 @@
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad          (when, unless)
+import Data.Monoid            ((<>))
+import Data.Text              (Text)
+import Options.Applicative    ((<**>))
+import Path                   (Path, Rel, File, Abs, Dir)
+import System.FilePath        (FilePath, (</>))
+import Turtle                 ((<|>))
+
+
+
+-----------------------
+-- === Constants === --
+-----------------------
+
+projectYaml :: Path Rel File
+projectYaml = $(Path.mkRelFile "./stack.yaml")
+
+releaseOpts :: [Text]
+releaseOpts = ["-fno-omit-interface-pragmas"]
+
+pkgBaseDirFromLocalBin :: FilePath
+pkgBaseDirFromLocalBin = "../../"
+
+stdlibSrcFolderLoc :: Turtle.FilePath
+stdlibSrcFolderLoc = "stdlib/Std/"
+
+licenseSrcFolderLoc :: Turtle.FilePath
+licenseSrcFolderLoc = "package/data/licenses"
+
+dataDistLoc :: Turtle.FilePath
+dataDistLoc = "config/env/"
+
+
+
+-------------------------
+-- === CLI Options === --
+-------------------------
+
+-- === Definition === --
+
+data CommandOpts = CommandOpts
+    { _releaseMode :: Bool
+    , _verbose     :: Bool
+    , _check       :: Bool
+    , _cleanStack  :: Bool
+    } deriving (Eq, Show)
+makeLenses ''CommandOpts
+
+
+-- === API === --
+
+optionsParser :: Options.Parser CommandOpts
+optionsParser = CommandOpts
+    <$> Options.switch (Options.long "release"
+        <> Options.help "Build the Luna package in release mode")
+    <*> Options.switch (Options.long "verbose"
+        <> Options.help "Enable verbose mode while building.")
+    <*> Options.switch (Options.long "check"
+        <> Options.help "Check that the build script itself builds.")
+    <*> Options.switch (Options.long "clean-stack"
+        <> Options.help "Clean stack build artefacts before build.")
+
+
+
+----------------------
+-- === Main API === --
+----------------------
+
+runBuilder :: CommandOpts -> Turtle.Shell ()
+runBuilder opts = when (not $ opts ^. check) $ do
+    -- Find the dist folder location
+    distributionFolder <- getDistFolder projectYaml
+
+    -- Clean Stack Build State (in Subshell)
+    when (opts ^. cleanStack) . Turtle.sh $ do
+        logMessage "Cleaning stack build artefacts"
+        Turtle.void $ Turtle.inprocWithErr "stack" ["clean"] Turtle.empty
+
+    -- Clean Package Build State
+    logMessage "Cleaning package dist"
+    pkgRootPath <- getPkgRoot distributionFolder pkgBaseDirFromLocalBin
+
+    pkgRootExists <- liftIO . Directory.doesDirectoryExist
+        $ Turtle.encodeString pkgRootPath
+
+    when pkgRootExists $ Turtle.rmtree pkgRootPath
+
+    -- Build and Copy Luna (in Subshell)
+    let buildMsg = "Building Luna in " <> if opts ^. releaseMode
+            then "release mode"
+            else "develop mode"
+    logMessage buildMsg
+
+    let ghcOpts   = if opts ^. releaseMode then releaseOpts else []
+        buildOpts = genBuildOpts ghcOpts
+
+    Turtle.sh $ Turtle.inprocWithErr "stack" buildOpts Turtle.empty >>= \out ->
```

Comment:
It'll now barf an error message, but the streaming nature of turtle isn't well suited to storing the output and barfing in case of an exception, so it tells you to rerun with `--verbose`. We can do better in future when we have a proper support library for this stuff.